File: Algorithm/C++/1000+/1037. Valid Boomerang.cc
/*
 * Created on Sun May 05 2019 17:9:47
 * Author: WuLC
 * EMail: liangchaowu5@gmail.com
 */

#include <vector>

using std::vector;

class Solution {
 public:
  bool isBoomerang(vector<vector<int>>& points) {
    vector<int> p1 = points[0], p2 = points[1], p3 = points[2];
    return (p2[1] - p1[1]) * (p3[0] - p2[0]) != (p2[0] - p1[0]) * (p3[1] - p2[1])
  }
};

File: Algorithm/Python/1000+/1037. Valid Boomerang.py
# -*- coding: utf-8 -*-
# Created on Sun May 05 2019 15:46:55
# Author: WuLC
# EMail: liangchaowu5@gmail.com

class Solution(object):
    def isBoomerang(self, points):
        """
        :type points: List[List[int]]
        :rtype: bool
        """
        p1, p2, p3 = points
        return (p2[1] - p1[1]) * (p3[0] - p2[0]) != (p2[0] - p1[0]) * (p3[1] - p2[1])